packages/@sanity/core/src/actions/deploy/undeployAction.js

```diff
@@ -0,0 +1,51 @@
+export default async (args, context) => {
+  const {apiClient, chalk, output, prompt} = context
+
+  const client = apiClient({
+    requireUser: true,
+    requireProject: true
+  })
+
+  // Check that the project has a studio hostname
+  let spinner = output.spinner('Checking project info').start()
+  const project = await client.projects.getById(client.config().projectId)
+  const studioHostname = project && (project.studioHostname || project.studioHost)
+  spinner.succeed()
+
+  if (!studioHostname) {
+    output.print('Your project has not been assigned a studio hostname.')
+    output.print('Nothing to undeploy.')
+    return
+  }
+
+  // Double-check
+  output.print('')
+
+  const shouldUndeploy = await prompt.single({
+    type: 'confirm',
+    default: false,
+    message: `
+https://${chalk.yellow(studioHostname)}.sanity.studio will be available for anyone to claim.
+  Are you ${chalk.red('sure')} you want to undeploy?`.trim()
+  })
+
+  if (!shouldUndeploy) {
+    return
+  }
+
+  const projectId = client.config().projectId
+  const uri = `/projects/${projectId}`
+
+  spinner = output.spinner('Undeploying studio').start()
+  try {
+    await client.request({uri, method: 'PATCH', body: {studioHost: null}})
+    spinner.succeed()
+  } catch (err) {
+    spinner.fail()
+    throw err
+  }
+
+  output.print(
+    'Studio undeployed. It might take a few minutes for the contents of the studio to be fully cleared.'
```

Comment:
"No longer available" and "might take a few minutes to be cleared" sounds weird.
I've changed it to:
> Studio undeploy scheduled. It might take a few minutes before ${url} is unavailable.